hdat/hdat_cli.py

```diff
@@ -76,49 +78,100 @@ def hdat_cli(arguments, suites, golden_store, archive, git_info):
             raise AbortError(_format_cases_status(cases_status))
     elif args.command == 'show':
         if not args.resultspec:
-            results = resolve_resultspecs(archive, args.resultspec)
+            results = resolve_resultspecs(archive, suites, args.resultspec)
             for result in results:
-                show_result(suites, result)
+                exit_status = show_result(suites, result)
+                if exit_status == 1:
+                    exit_code = 1
         for resultspec in args.resultspec:
-            results = resolve_resultspecs(archive, resultspec)
-            for result in results:
-                show_result(suites, result)
+            results = resolve_resultspecs(archive, suites, resultspec)
```

Comment:
I may be missing something, but it seems that every time `resolve_resultspecs` returns a string, we do the same thing---print the error and return with exit_code 1.  I think this is better handled by an exception.  (note that error messages should probably go to stderr not stdout)

Functions should almost always return only one type.  I.e., it is pretty rare (and usually bad) to return a string OR a list of results.  I think it is better for the function to return a list of results OR raise an exception.